subsys/net/lib/lwm2m/lwm2m_engine.c

```diff
@@ -4699,6 +4700,17 @@ int lwm2m_socket_start(struct lwm2m_ctx *client_ctx)
 			lwm2m_engine_context_close(client_ctx);
 			return -errno;
 		}
+
+		if (client_ctx->desthostname != NULL &&
+				client_ctx->desthostnamelen <= URI_MAX_LEN) {
```

Comment:
I don't think it's needed to verify the length, socket implementation does not put a limit on that (as long as there's enough mem on heap, but that we are not able to verify here).